build/buildkit/buildimage/build_image.go

```diff
@@ -62,6 +64,16 @@ func (b BuildImage) ExecBuild(ctx Context) error {
 	} else {
 		execCtx = buildinstruction.NewExecContextWithoutCache(ctx.BuildType, ctx.BuildContext, b.LayerStore)
 	}
+	//judge the docker engine version
+	dockerServerVersion, err := b.DockerClient.GetServerVersion()
```

Comment:
Check the version in `ImagesCacheToRegistry`, so you don't need the bool argument